### PR TITLE
Allow for `decimal` type descriptor

### LINKED ISSUE #766: Allow `decimal` as a type descriptor
Support decimal in var-decl.
Eg: `decimal x = 10d;`

Part of #289

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
Fixes #766

### CODE REVIEW

File: compiler/testSuite/11-decimal/optinal1-v.bal [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import ballerina/io;
```
Comment (jclark) [inline compiler/testSuite/11-decimal/optinal1-v.bal]: Fix spelling of filename of this test.

**Review by jclark — CHANGES_REQUESTED** [review]
Test cases need to be in 12-decimal.

File: compiler/testSuite/11-decimal/optinal1-v.bal (hunk 1 above)
Comment (KavinduZoysa) [reply compiler/testSuite/11-decimal/optinal1-v.bal]: Changed

**Comment by KavinduZoysa** [comment]
> Test cases need to be in 12-decimal.

Changed

**Pull request merged**